### PR TITLE
[AJ-1182] Return errored apps from resolveWdsApp

### PR DESCRIPTION
**Pull request by nawatts**
https://broadworkbench.atlassian.net/browse/AJ-1182

Currently, `resolveWdsApp` only returns "healthy" (running, provisioning, stopping, or stopped) apps. This limits the usefulness of the WDS troubleshooter, which will show the WDS app status as "unknown" even when the app has errored / failed to start.

This changes `resolveWdsApp` to return apps of any status, prioritized as before. The impact of this change is limited because `resolveWdsApp` is only used in two places: the WDS Troubleshooter and `resolveWdsUrl`. `resolveWdsUrl`, which is more broadly used, only returns a URL if the app's status is running.

## Before
Screenshot 2023-07-21 at 11 36 50 AM

## After  
Screenshot 2023-07-21 at 12 01 06 PM

## Testing

The prioritization of apps is covered by unit tests for `resolveWdsUrl`. 

You can override the list apps response to return an errored app with:
```js
ajaxOverridesStore.set([
  {
    filter: { url: /\/apps\/v2\// },
    fn: ajaxOverrideUtils.makeSuccess([
        {
            "workspaceId": "eeff20de-0070-4c1c-809c-6f0c2e01eb24",
            "cloudContext": {
                "cloudProvider": "AZURE",
                "cloudResource": "0cb7a640-45a2-4ed6-be9f-63519f86e04b/0fa61a72-3a6b-4d51-a7e9-75fbeb7c7ed9/mrg-terra-dev-previ-20230626173007"
            },
            "kubernetesRuntimeConfig": {
                "numNodes": 1,
                "machineType": "Standard_A2_v2",
                "autoscalingEnabled": false
            },
            "errors": [],
            "status": "ERROR",
            "proxyUrls": {
                "wds": "https://lz81067ca761368f63128f883248bc7c26e250eaa41f49a163.servicebus.windows.net/wds-eeff20de-0070-4c1c-809c-6f0c2e01eb24-eeff20de-0070-4c1c-809c-6f0c2e01eb24/"
            },
            "appName": "wds-eeff20de-0070-4c1c-809c-6f0c2e01eb24",
            "appType": "WDS",
            "diskName": null,
            "auditInfo": {
                "creator": "user@example.com",
                "createdDate": "2023-07-20T18:42:28.798500Z",
                "destroyedDate": null,
                "dateAccessed": "2023-07-20T18:42:28.798500Z"
            },
            "accessScope": "WORKSPACE_SHARED",
            "labels": {}
        }
    ])
  }
])
```

### CODE REVIEW

**Review by calypsomatic — APPROVED** [review]

File: src/libs/ajax/data-table-providers/WdsDataTableProvider.ts [hunk 1]
```diff
@@ -90,48 +91,35 @@ const getRelationParts = (val: unknown): string[] => {
...
+      (app) => (app.appType === 'WDS' ? -1 : 1),
+      // Prefer running apps
+      (app) => (app.status === 'RUNNING' ? -1 : 1),
+      // Prefer healthy apps
+      (app) => (healthyStates.includes(app.status) ? -1 : 1),
+      // Prefer apps created earlier
+      (app) => new Date(app.auditInfo?.createdDate).valueOf(),
+    ],
+    eligibleApps
+  );
```
Comment (davidangb) [inline src/libs/ajax/data-table-providers/WdsDataTableProvider.ts:117]: I don't think this logic is correct, just from reading the code; I did not try to write unit tests or test it empirically.

This should never, under any circumstance, return an app that is anything other than `appType === 'WDS'`. It looks like this logic allows that, in a corner case.

Comment (nawatts) [reply src/libs/ajax/data-table-providers/WdsDataTableProvider.ts:117]: I was basing this on the current implementation, which can return an app with `appType === 'CROMWELL'`, though it prefers WDS.

This can be seen in the unit tests for `resolveWdsUrl`.
https://github.com/DataBiosphere/terra-ui/blob/35ddbc59e4e6f5c587ca45a34e0a734bf3fb80f3/src/libs/ajax/data-table-providers/WdsDataTableProvider.test.ts#L770-L892

**Comment by nawatts** [comment]
Rebased to remove `ts-expect-error` comments after #4060 merged.

File: src/libs/ajax/data-table-providers/WdsDataTableProvider.ts (hunk 1 above)
Comment (davidangb) [reply src/libs/ajax/data-table-providers/WdsDataTableProvider.ts:117]: Ah - I think I understand what's going on here, this is support for the much older `CROMWELL` apps where WDS and CBAS were still bundled. I suppose we leave it in for now, but at some point soon we should remove that logic and stop supporting super-old deployments.

**Review by davidangb — APPROVED** [review]
